vspec/model/vsstree.py

```diff
@@ -259,6 +259,9 @@ def get_struct_qualified_name(self, struct_name) -> str:
             root = path
 
         # find the struct node under the branch
+        if root is None:
```

Comment:
When would this trigger? properties are always defined under struct nodes and we have an orphan check for that.

Reply:
This a typical checker "false positive" - it is not smart enough to know if `root` actually will be None here or not. In this case it it is mypy that complains: 

`error: Item "None" of "Optional[VSSNode]" has no attribute "children"  [union-attr]`

It seems that the reason for this error is that we assign `root = None` before the while loop. I guess that indicates to mypy that we actually not always may find a root. If we remove that line mypy will not complain.

```
        # find the ancestor branch
        root = None
        while not path.is_branch():
            path = path.parent
            root = path
```
But I guess it is also a question on how "bullet-proof" we want the methods to be in class  VSSNode. Do we want them to not crash regardless of how "self" looks like, or do we expect them to be used only in specific context.

Here I can see a few alternatives to make mypy happy.

- Keep the PR as is, i.e. let method return None if "self" has no parent (even if that never should happen as we use the method today)
- Remove the `root = None` assignment. That will mypy happy, but you will get a crash if "self" has no parent (even if that shall not happen as we use the method today)
- Add an `# type: ignore[union-attr]` on line 265. Will make mypy happy, but we will still get a crash if there is no parent. 

I prefer to either have it bullet-prof (i.e. keep as in PR) or remove the assignment and possibly add a comment that we expect there to always be a parent.

Reply:
In another PR, we need to make the callers check if the returned value is `None`